lib/open_pantry/food_selection.ex

```diff
@@ -72,14 +72,25 @@ defmodule OpenPantry.FoodSelection do
     stock_distribution = StockDistribution.find_or_create(package.id, stock.id)
     cost = stock.credits_per_package
     Multi.new
-    |> Multi.update_all(:stock_distribution, StockDistribution.query(stock_distribution.id), [inc: [quantity: quantity]])
     |> Multi.update_all(:stock, Stock.query(stock.id), [inc: [quantity: -quantity]])
     |> deduct_credits(cost, quantity, type_id, user_id, {stock.meal_id, stock.offer_id, stock.food_id })
-    |> Multi.run(:validate_stock_distribution, StockDistribution, :validate_stock_distribution, [stock_distribution, quantity])
+    |> update_stock_distribution(stock_distribution, quantity)
     |> Repo.transaction
     stock_distribution
   end
 
+  @spec update_stock_distribution(Ecto.Multi.t, StockDistribution.t, integer() ) :: Ecto.Multi.t
```

Comment:
Excellent, thank you for adding a Dialyzer spec!  I have another issue looking for help with getting these more pervasive and useful in the app, do you have any experience with Dialyzer?  Ideally I'd also like to learn more about how to understand it and do it myself, and perhaps make some PR's the Dialyxir to give more readable/useful error messages (I did something similar for ex_crypto when I found Erlang's error behavior in :crypto module confusing), but depending on your time and aptitude, anything is appreciated!

Reply:
Sadly, I do not have experience with Dialyzer, but I also want to learn it.
I might look into the issue if I have time.